soda_machine.rb

```diff
@@ -1,18 +1,46 @@
+# require ("soda.rb")
 class SodaMachine
   attr_reader :sodas, :cash
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @sold_soda = nil
+    @brand = []
+    @index = 0
   end
 
   def current_inventory_count
+    #return no of sodas
+    num_of_soda = @sodas.count
   end
 
```

Comment:
this implementation of `find_soda` is a little long.  seems more complicated than it needs to be.

your goal is to find a soda that matches an incoming (parameter) `soda_brand` value.  and you have a list of `@sodas` in an instance variable.

just loop over those sodas looking for a matching brand and, if you find one, return the matching `soda` object.

so, something like this:

```ruby
def find_soda(soda_brand)
  @sodas.each do |soda|
    return soda is soda.brand == soda_brand
  end
end
```